### DISCUSSION TITLE
how to obtain optimal RSF

### DISCUSSION THREAD

**Discussion by wymin576** [kogalur/randomForestSRC#426] [Q&A] [answered]
I have a question. When performing random survival forest (RSF), how to obtain optimal RSF? My idea is to  screen variables  using variable importance (VIMP), then tune model parameters: ntree, nodesize and nsplit.
i am not sure whether the functions tune.nodesize and tune.rfsrc can meet the requirement for the second step.

The codes for the first step are as follows,
library(randomForestSRC)
data(peakVO2, package = "randomForestSRC")
dta <- peakVO2
obj <- rfsrc(Surv(ttodead,died)~., dta,
             ntree = 1000, nodesize = 5, nsplit = 50, importance = TRUE)
jk.obj <- subsample(obj)
plot(jk.obj)

Positive VIMP indicates variables is helpfu to predict survival time/event while zero or negative values identify noise variables. As a result, we only keep the variable with positive VIMP.
Is my strategy right? Thanks in advance.

**Comment by ishwaran** [comment, marked as answer]
This is a low-dimensional problem therefore it is unnecessary to pre-filter variables prior to fitting the RSF (which will introduce bias into your estimated prediction error).  I would suggest that if you want to perform some tuning then it would be enough to tune on `nodesize` (`nsplit` and especially `ntree` is excessive).

**Reply by wymin576** [reply]
Thank you for the fast reply!
